cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,47 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('/');
+    cy.task('generateUser').then((genereteUser) => {
+      user = genereteUser;
+    });
   });
 
-  it('', () => {
-
+  it('Should register new user', () => {
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
+    cy.get('#userEmail')
+      .type(user.email);
+    cy.contains('.custom-control-label', user.gender)
+      .click();
+    cy.findByPlaceholder('Mobile Number')
+      .type(user.mobileNumber);
+    cy.get('#dateOfBirthInput')
+      .type(`{selectall}23 Feb 1991{Enter}`);
+    cy.get('.subjects-auto-complete__value-container')
+      .type('en{enter}' + 'phy{enter}');
+    cy.contains('.custom-control-label', user.hobby)
+      .click();
+    cy.findByPlaceholder('Current Address')
+      .type(user.address);
+    cy.get('#state')
+      .type('{downarrow}{enter}');
+    cy.get('#city')
+      .type('{downarrow}{enter}');
+    cy.get('#submit').click();
+    cy.contains('tr', user.firstName).should('be.visible');
```

Comment:
I would recommend writing more concrete checks, not just if the element is visible, but if some selector contains some data (like you did for State and City)
Fix it for all further checks, please

Reply:
Fixed!